tests/test_behavior/test_external_video_interface.py

```diff
@@ -0,0 +1,306 @@
+import shutil
+import tempfile
+import unittest
+from copy import deepcopy
+from datetime import datetime
+from pathlib import Path
+
+import numpy as np
+from dateutil.tz import gettz
+from hdmf.testing import TestCase
+from pynwb import NWBHDF5IO
+
+from neuroconv import NWBConverter
+from neuroconv.datainterfaces.behavior.video.externalvideointerface import (
+    ExternalVideoInterface,
+)
+from neuroconv.utils import dict_deep_update
+
+try:
+    import cv2
+
+    skip_test = False
+except ImportError:
+    skip_test = True
+
+
+@unittest.skipIf(skip_test, "cv2 not installed")
+class TestExternalVideoInterface(TestCase):
+    def setUp(self) -> None:
+        self.test_dir = Path(tempfile.mkdtemp())
+        self.video_files = self.create_videos()
+        self.nwb_converter = self.create_video_converter()
+        self.metadata = self.nwb_converter.get_metadata()
+        self.metadata["NWBFile"].update(session_start_time=datetime.now(tz=gettz(name="US/Pacific")))
+        self.nwbfile_path = self.test_dir / "video_test.nwb"
+        self.aligned_segment_starting_times = [0.0, 50.0]
+
+    def tearDown(self) -> None:
+        shutil.rmtree(self.test_dir)
+        del self.nwb_converter
+
+    def create_videos(self):
```

Comment:
This is one of the slowest part of our testing suit. 

I feel that it does not add anything more than the tests on data that we have and if that is true I would prefer to have that one only.

If it does add something that I am not aware off we should make it a fixture for the three interfaces so it is reused.

Reply:
To be more clear, I claim that everything on this tests is or should be covered here:
https://github.com/catalystneuro/neuroconv/blob/2f9661ff4c3920b39c5eef2bce8063751390784f/tests/test_on_data/behavior/test_behavior_interfaces.py#L259-L272

Reply:
I'm confused, why would you ask to refactor the existing tests to pytest style if you think they should just be completely removed in favor of these test_on_data tests?

Reply:
And more to the point, only test_video_interface.py tests error handling, irregular timestamps, and multiple video files in the same interface, so it's not really redundant.

Reply:
Plus, this is stretching the scope of this PR. Updating the testing structure to focus more on test_on_data should be it's own PR.

Reply:
> I feel that it does not add anything more than the tests on data that we have and **if that is true** I would prefer to have that one only.

> **If it does add something that I am not aware off** we should make it a fixture for the three interfaces so it is reused.

It seems that the tests that create data on the fly do cover some aspects that are not covered on the tests with data from gin. If that is the case, then I am fine for adding new tests. 

But given that we are adding new tests, do it with a modular fixture so the computation is not repeated.

Reply:
Ok, it is a reused fixture now.  But more importantly, I reduced the video size from (30, 640, 480) to (30, 64, 48), which speeds things up dramatically.

Reply:
Even better. Thanks.